src/vellum/workflows/state/tests/test_context.py

```diff
@@ -0,0 +1,82 @@
+from vellum.workflows.state.context import WorkflowContext
+from vellum.workflows.utils.uuids import uuid4_from_hash
+
+
+def test_resolve_workflow_deployment__returns_workflow_with_generated_files():
+    """
+    Test that resolve_workflow_deployment returns a workflow with artifacts
+    in generated_files using the correct prefix format.
+
+    GIVEN a deployment name and release tag
+    WHEN we resolve the workflow deployment
+    THEN the method should return a workflow with artifacts in the correct prefix location
+    """
+    # GIVEN a deployment name and release tag
+    deployment_name = "test_deployment"
+    release_tag = "v1.0.0"
+
+    expected_hash = str(uuid4_from_hash(f"{deployment_name}_{release_tag}"))
+    expected_prefix = f"vellum_workflow_deployment_{expected_hash}"
+
+    # Create a simple test node for the resolved workflow
+    test_node_code = """
+from vellum.workflows.nodes.displayable import TemplatingNode
+from vellum.workflows.outputs import BaseOutputs
+
+class TestNode(TemplatingNode):
```

Comment:
Extend from BaseNode instead and have a simple run method that returns `self.Outputs(result="Hello, {template}")`